src/lib.rs

```diff
@@ -2950,6 +2951,124 @@ fn digraph_transitivity(graph: &digraph::PyDiGraph) -> f64 {
     }
 }
 
+pub fn _core_number<Ty>(
+    py: Python,
+    graph: &StableGraph<PyObject, PyObject, Ty>,
+) -> PyResult<PyObject>
+where
+    Ty: EdgeType,
+{
+    let node_num = graph.node_count();
+    if node_num == 0 {
+        return Ok(PyDict::new(py).into());
+    }
+
```

Comment:
Since now the size of bin_boundaries depends on node_num, an early return is added to prevent the algorithm from failing from an empty graph. Let me know if there are better ways.